### ISSUE TITLE
CH 4 Mini Batch GD with Scikit-Learn (small addition)

### ISSUE THREAD

**Issue by nyilmaz1** [ageron/handson-ml#226]
Hello @ageron,

Great book, learning more and more in each chapter! I see that you write mini-batch Gradient Descent manually with defining the mini-batch size (Ch 4-Jupyter notebook line 23), which is very helpful to understand what's happening under the hood. I just wanted to write here how to do it with Scikit-Learn in case anyone wonders and searches it here:
``` 
mini_batchGD = sgd_reg.partial_fit(X,y)
mini_batchGD.intercept_, mini_batchGD.coef_
```
The output, as expected, is similar to what I had with SGD regressor.

Best wishes,
Nevin

**Comment by ageron** [comment]
Good point, thanks Nevin.

The `partial_fit()` method performs a single step of mini-batch gradient descent, so you would need to split the training set into multiple mini batches and call `partial_fit()` repeatedly, like this:

```python
for epoch in range(n_epochs):
    for iteration in range(n_iterations):
        X_batch, y_batch = ...  # get the next mini batch
        sgd_reg.partial_fit(X_batch, y_batch)
```

Hope this helps,
Aurélien

**Comment by nyilmaz1** [comment]
Thanks @ageron, this is very helpful.

Best,
Nevin
